File: java/src/test/java/banking/AccountControllerTest.java
package banking;

import org.junit.Test;

import static org.hamcrest.CoreMatchers.equalTo;
import static org.junit.Assert.*;
import static org.mockito.Mockito.mock;

public class AccountControllerTest {

    private Account account = mock(Account.class);
    private AccountController accountController =
            new AccountController(() -> account);

    @Test
    public void name_is_totally_TODO() {
        // setup mock for account here

        // make a call to controller

        // verify the result is correct




        // just verifying that the test runner works
        assertThat(2 + 3, equalTo(4));
    }

}

File: kotlin/src/test/kotlin/banking/AccountControllerTest.kt
package banking

import org.hamcrest.CoreMatchers
import org.junit.Assert
import org.junit.Test
import org.mockito.Mockito.mock

class AccountControllerTest {

    private val account = mock(Account::class.java)
    private val accountController = AccountController { account }

    @Test
    fun `name is totally TODO`() {
        // setup mock for account here

        // and make call to controller

        // check the result



        // just verifying that the test runner works
        Assert.assertThat(2 + 3, CoreMatchers.equalTo(4))
    }

}